src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,51 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    private function getDatabaseBackedField(string $fieldPath): ?string
+    {
+        $component = $this;
+        $fieldParts = [];
+
+        foreach ($parts = explode('.', $fieldPath ?? '') as $nextPart) {
```

Comment:
```suggestion
        $parts = explode('.', $fieldPath ?? '');
        foreach ($parts as $nextPart) {
```
Please avoid assigning variables inside a foreach clause.

Reply:
Updated